utils/stringforwarder/stringforwarder_test.go

```diff
@@ -106,27 +107,28 @@ func (*stringForwarderSuite) TestMessagesDroppedWhenBusy(c *gc.C) {
 	c.Check(dropCount, gc.Equals, uint64(3))
 }
 
-func (*stringForwarderSuite) TestRace(c *gc.C) {
-	forwarder := stringforwarder.NewStringForwarder(noopCallback)
+func (*StringForwarderSuite) TestRace(c *gc.C) {
+	forwarder := stringforwarder.New(noopCallback)
 	stop := make(chan struct{})
-	go func() {
-		// In 100ms we can hit 1M calls, so make this 10M, and make the
-		// timeout 1ms, that gives us a safety factor of 1000.
-		for i := 0; i < 10*1000*1000; i++ {
+	wg := &sync.WaitGroup{}
+	f := func(wg *sync.WaitGroup) {
+		wg.Done()
+		for {
 			select {
 			case <-stop:
 				return
 			default:
-				forwarder.Receive("next message")
+				forwarder.Forward("next message")
 			}
 		}
-		c.Errorf("managed to send too many messages without being stopped")
```

Comment:
This is the actual fix.... removing the limited length for-loop. It's not really testing anything important.